tests/tensor/test_nlinalg.py

```diff
@@ -533,16 +536,15 @@ def test_wrong_rcond_dimension(self):
 
 
 class TestMatrix_power:
-    def test_numpy_compare(self):
+    @pytest.mark.parametrize("n", [0, 1, 2, 3, 4, 5, 11])
+    def test_numpy_compare(self, n):
         rng = np.random.RandomState(utt.fetch_seed())
-        A = tensor.matrix("A", dtype=theano.config.floatX)
-        Q = matrix_power(A, 3)
-        fn = function([A], [Q])
         a = rng.rand(4, 4).astype(theano.config.floatX)
```

Comment:
There's really no need to use random numbers in these examples.  We're better off using a single representative 2 x 2 matrix for these tests (e.g. one that's neither idempotent nor vanishes too quickly).

Reply:
Well, the way I approached this is basically by making minimal changes to the codebase.

Reply:
My comment was actually about big-picture changes that I would like to start making in Theano.  Your approach is definitely the correct one, so no problems there.